frmts/gtiff/geotiff.cpp

```diff
@@ -1132,6 +1132,84 @@ CPLString GTiffGetCompressValues(bool &bHasLZW, bool &bHasDEFLATE,
     return osCompressValues;
 }
 
+/************************************************************************/
+/*                    OGRGTiffDriverGetSubdatasetInfo()                 */
+/************************************************************************/
+
+struct OGRGTiffDriverSubdatasetInfo : public GDALSubdatasetInfo
+{
+  public:
+    explicit OGRGTiffDriverSubdatasetInfo(const std::string &fileName)
+        : GDALSubdatasetInfo(fileName)
+    {
+    }
+
+    // GDALSubdatasetInfo interface
+  private:
+    void parseFileName() override
+    {
+        if (!STARTS_WITH_CI(m_fileName.c_str(), "GTIFF_DIR:"))
+        {
+            return;
+        }
+
+        char **papszParts{CSLTokenizeString2(m_fileName.c_str(), ":", 0)};
```

Comment:
you can use CPLStringList to avoid having to explicitly free
```suggestion
        CPLStringList aosParts(CSLTokenizeString2(m_fileName.c_str(), ":", 0));
```